metrics/tenant_registries.go

```diff
@@ -443,53 +461,37 @@ func sum(mf *dto.MetricFamily, fn func(*dto.Metric) float64) float64 {
 	return result
 }
 
-// max returns the maximum value from all metrics from same metric family (= series with the same metric name, but different labels)
-// Supplied function extracts value.
-func max(mf *dto.MetricFamily, fn func(*dto.Metric) float64) float64 {
-	return fold(mf, fn, func(val, res float64) float64 {
-		if val > res {
-			return val
-		}
-		return res
-	})
-}
-
-// min returns the minimum value from all metrics from same metric family (= series with the same metric name, but different labels)
-// Supplied function extracts value.
-func min(mf *dto.MetricFamily, fn func(*dto.Metric) float64) float64 {
-	return fold(mf, fn, func(val, res float64) float64 {
-		if val < res {
-			return val
-		}
-		return res
-	})
-}
-
 // fold returns value computed from multiple metrics, using folding function. if there are no metrics, it returns 0.
 func fold(mf *dto.MetricFamily, fn func(*dto.Metric) float64, foldFn func(val, res float64) float64) float64 {
 	result := math.NaN()
 
 	for _, m := range mf.GetMetric() {
 		value := fn(m)
+		if math.IsNaN(value) {
+			continue
+		}
 		if math.IsNaN(result) {
 			result = value
 		} else {
 			result = foldFn(value, result)
 		}
 	}
 
-	// If there's no metric, we do return 0 which is the gauge default.
-	if math.IsNaN(result) {
-		return 0
-	}
-
 	return result
 }
 
 // This works even if m is nil, m.Counter is nil or m.Counter.Value is nil (it returns 0 in those cases)
 func counterValue(m *dto.Metric) float64 { return m.GetCounter().GetValue() }
 func gaugeValue(m *dto.Metric) float64   { return m.GetGauge().GetValue() }
 
+// gaugeValueNaN returns Gauge value or math.NaN.
+func gaugeValueNaN(m *dto.Metric) float64 {
```

Comment:
[nit] I would clarify it a bit.

```suggestion
func gaugeValueOrNaN(m *dto.Metric) float64 {
```

Reply:
Done in https://github.com/grafana/dskit/pull/368/commits/661d25d7e6718d7ce9fa62e69c82b6b8c6957cb7.